Blockzilla/URLBar.swift

```diff
@@ -713,11 +714,27 @@ extension URLBar: AutocompleteTextFieldDelegate {
     }
 
     func autocompleteTextFieldShouldReturn(_ autocompleteTextField: AutocompleteTextField) -> Bool {
+        // Checks if the user used the autocomplete to finish their query
+        let domains = domainCompletion.getDomains()
+        if let autocompleteText = autocompleteTextField.text, autocompleteText != currentUserInput {
+            for domain in domains {
+                if domain == autocompleteText.split(separator: "/")[0] {
+                    Telemetry.default.recordEvent(TelemetryEvent(category: TelemetryEventCategory.action, method: TelemetryEventMethod.click, object: TelemetryEventObject.autocomplete))
+
+                    // Just in case a user inputs a duplicate url in the custom autocomplete list
+                    break
+                }
+            }
+        }
```

Comment:
I am not familiar with this functionality enough to know why this method is a confirmation of autocompletion. Is there no deterministic way that doesn't involve a heuristic?